app/assets/javascripts/controllers.js

```diff
@@ -174,6 +160,18 @@ angular.module('HNSearch.controllers', ['ngSanitize', 'ngDropdowns', 'pasvaz.bin
     }
   };
 
+  $scope.clickHit = function(objectID, position) {
+    aa && aa('click', {
+      objectID: objectID, position: position
+    });
+  };
+
+  $scope.gotoHit = function(objectID) {
+    aa && aa('conversion', {
+      objectID: objectID
+    });
```

Comment:
@JonasBa this will be done right before the user gets actually redirected to another page. How can we ensure the browser has enough time to do the XHR before redirecting?

Reply:
The click and conversion requests use the sendBeacon API if it is available. If not, it default to a synchronous XHR.  That said, there's a bit of a tradeoff here to be made between making sure the call is sent and degrading the navigating experience or faster experience but we might miss some event. 

The other possible alternative is save the request to localStorage and retry sending next time script loads instead of doing a synchronous req. Afaik, @sfriquet drops queries after 1h, so it could be a possible alternative if we decide that dropping some requests is ok.